src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala

```diff
@@ -93,7 +93,7 @@ class OfferOperationFactory(
       Mesos.Resource.newBuilder()
         .setName("disk")
         .setType(Mesos.Value.Type.SCALAR)
-        .setScalar(Mesos.Value.Scalar.newBuilder().setValue(vol.persistentVolume.persistent.size.toDouble).build())
+        .setScalar(Mesos.Value.Scalar.newBuilder().setValue(vol.persistentVolume.persistent.size.get.toDouble).build())
```

Comment:
Don't use `get` but preferable `foreach` around adding the disk resource. It is optional, correct?

If it is not optional, make sure we get a more meaningful error message by throwing an appropriate exception.